### DISCUSSION TITLE
TraceId and SpanId not coming during manual instrumentation

### DISCUSSION THREAD

**Discussion by omsaichetanbe** [open-telemetry/opentelemetry-java-instrumentation#11611] [Q&A]
Hi, 

I have been exporting logs and traces to the Grafana but when trying to open tempo and check the logs for a particular traceId, it is not showing any logs after adding correlation. 

I have instrumented a Java Azure Function Application with OpenTelemetry Java Agent Manually and my understanding is that the TraceId and SpanId are propagated and added from the agent. When doing Span.current().getSpanContext().getTraceId() I am seeing 00000000000000000000000000000000. 

I have tried adding the logback.xml
![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/125671419/ccc9faae-d1dc-40a6-86be-f07b21c6c0cf)

and the dependency :
<dependency>
		    <groupId>io.opentelemetry.instrumentation</groupId>
		    <artifactId>opentelemetry-logback-mdc-1.0</artifactId>
		    <version>2.4.0-alpha</version>
  		</dependency>
 
 Any help would be much appreciated.

**Comment by omsaichetanbe** [comment]
I also noticed that only few time the application is generating the traceid and spanid and few other time traceid and spanid are not added to the logs. 

When the Opentelemetry java agent jar is adding the traceid and spanid for few requests it only added for few of the logs not the other. I am confused here with when it is adding and when it is not adding the traceid and spanid.

**Comment by trask** [comment]
hi @omsaichetanbe, the OpenTelemetry Java agent doesn't have automatic context propagation for Azure Functions

there are two approaches forward here (both are good):
* add automatic context propagation for Azure Functions to the OpenTelemetry Java agent (this probably wouldn't be too hard to copy over the relevant bits from https://github.com/microsoft/ApplicationInsights-Java/tree/main/agent/instrumentation/azure-functions)
* open an issue in https://github.com/Azure/azure-functions-java-worker as I believe the Azure Functions team is planning to support OpenTelemetry natively in Azure Functions (cc @RohitRanjanMS)

**Comment by omsaichetanbe** [comment]
Hi @trask , our main goal was not to make any code changes with manual instrumentation or auto instrumentation, we either need to send the traces, logs, metrics using manual instrumentation using java agent jar or just integrate the dependencies in the pom.xml and achieve integration of logs traces with traceId and SpanId. Is it possible to do so?

**Comment by omsaichetanbe** [comment]
@trask , can you please let me know how I can get traceId and SpanId exported from a springboot application.
